lib/state.js

```diff
@@ -90,15 +90,23 @@ export class State extends Map {
     return schemas;
   }
 
-  async build() {
+  async build(config) {
     const buildPlugins = [];
     if (this.has("extensions")) {
-      for (const buildPlugin of this.get("extensions").build) {
-        buildPlugins.push(...(await buildPlugin({ cwd: this.cwd, development: this.development })));
+      for (const buildFunction of this.get("extensions").build) {
+        let plugins = await buildFunction({
+          cwd: this.cwd,
+          development: this.development,
+          config,
+        });
+        if (!Array.isArray(plugins)) {
+          plugins = [plugins];
+        }
+        buildPlugins.push(...plugins);
       }
     }
     if (this.has("local") && typeof this.get("local").build === "function") {
-      buildPlugins.push(...(await this.get("local").build({ cwd: this.cwd, development: this.development })));
+      buildPlugins.push(...(await this.get("local").build({ cwd: this.cwd, development: this.development, config })));
```

Comment:
We will need to do the same with this one and handle both the array and non array cases same as you have done above

Reply:
good catch! thanks I've added it ❤️